fe/fe-core/src/main/java/org/apache/doris/load/ExportJob.java

```diff
@@ -266,17 +270,83 @@ private void generateQueryStmt() {
             }
         }
 
-        List<TableRef> tableRefList = Lists.newArrayList();
-        tableRefList.add(this.tableRef);
-        FromClause fromClause = new FromClause(tableRefList);
+        ArrayList<ArrayList<TableRef>> tableRefListPerQuery = splitTablets(stmt);
+        LOG.info("Export task is split into {} outfile statements.", tableRefListPerQuery.size());
+        for (int i = 0; i < tableRefListPerQuery.size(); i++) {
+            LOG.info("Outfile clause {} is responsible for tables: {}", i,
+                    tableRefListPerQuery.get(i).get(0).getSampleTabletIds());
+        }
+        for (ArrayList<TableRef> tableRefList : tableRefListPerQuery) {
+            FromClause fromClause = new FromClause(tableRefList);
+            // generate outfile clause
+            OutFileClause outfile = new OutFileClause(this.exportPath, this.format, convertOutfileProperties());
+            SelectStmt selectStmt = new SelectStmt(list, fromClause, this.whereExpr, null,
+                    null, null, LimitElement.NO_LIMIT);
+            selectStmt.setOutFileClause(outfile);
+            selectStmt.setOrigStmt(new OriginStatement(selectStmt.toSql(), 0));
+            selectStmtList.add(selectStmt);
+        }
+    }
+
+    private ArrayList<ArrayList<TableRef>> splitTablets(ExportStmt stmt) throws UserException {
+        // get tablets
+        Database db = Env.getCurrentEnv().getInternalCatalog().getDbOrAnalysisException(stmt.getTblName().getDb());
+        OlapTable table = db.getOlapTableOrAnalysisException(stmt.getTblName().getTbl());
+        List<Long> tabletIdList = Lists.newArrayList();
+        table.readLock();
+        try {
+            Collection<Partition> partitions = new ArrayList<Partition>();
+            // get partitions
+            // user specifies partitions, already checked in ExportStmt
+            if (partitionNames != null) {
+                for (String partName : partitionNames) {
+                    partitions.add(table.getPartition(partName));
+                }
+            } else {
+                partitions = table.getPartitions();
```

Comment:
A table may contains thousands of partitions. I think we need to config to limit the max number of partition, as a protection

Reply:
done